src/dashboard/src/components/advanced_search.py

```diff
@@ -59,7 +71,9 @@ def search_parameter_prep(request):
             except:
                 types.insert(index, '')
 
-            index = index + 1
+        for index, field in enumerate(fields):
```

Comment:
>  Are we overwriting entries in fields?

Yep! The idea is that there is a second array of "other fields", populated when the user has manually typed in a field. If the field name indicates that an "other field" has been chosen, the placeholder value is overwritten with the actual name of the field from the other array.

Because there can be multiple searches, this enumerates from the beginning of the array to the end.